src/App/Datums.cpp

```diff
@@ -302,12 +302,12 @@ void LocalCoordinateSystem::unsetupObject()
 {
     const auto& objsLnk = OriginFeatures.getValues();
     // Copy to set to assert we won't call method more then one time for each object
-    std::set<App::DocumentObject*> objs(objsLnk.begin(), objsLnk.end());
+    const std::set<App::DocumentObject*> objs(objsLnk.begin(), objsLnk.end());
     // Remove all controlled objects
     for (auto obj : objs) {
         // Check that previous deletes didn't indirectly remove one of our objects
-        const auto& objsLnk = OriginFeatures.getValues();
-        if (std::find(objsLnk.begin(), objsLnk.end(), obj) != objsLnk.end()) {
+        const auto& objsLnk2 = OriginFeatures.getValues();
+        if (std::ranges::find(objsLnk2, obj) != objsLnk2.end()) {
```

Comment:
It's too bad `std::ranges::contains` wasn't added in c++20 (it comes in c++23). Having that would clean up this pattern like:

```c++
if (std::ranges::contains(objsLnk2, obj)) {
```

However, you can use the member function `contains()` in associative containers (set, map etc., not available in `vector`) that was added in c++20.

By using the member functions it not only makes the code easier to read, it also reduces the complexity when checking if a value exists:
```c++
if (objsLnk2.contains(obj)) {
```

I would prefer if that is used where possible.